### ISSUE TITLE
Not updating on session_state change

### ISSUE THREAD

**Issue by coolkau** [sqlinsights/streamlit-toggle-switch#2]
I am new to steamlit and suprprised they dont have a a native toggle. Awesome somebody coded it up so I dont have to learn building componentes before building my first app.

One issue I found it does not update on state_session changes:

```
st_toggle_switch("test", label_after=True, default_value=True, key="toggle")

def callback_test():
    st.session_state["toggle"] = False

st.button('callback', on_click=callback_test)

```

This code works nicely on the inbuilt widgets, e.g. the select_slider, but your widget just remains at activated while the sesssion state is reset to false.

(Unrelated it would be great to have a callback. Once I know how components work I will happily suggest the changes)

**Comment by sqlinsights** [comment]
Good catch. I am fairly new at building Streamlit components but will have to check if callback is allowed on custom components as they render differently than native components.

**Comment by coolkau** [comment]
[Thats what I found](https://github.com/streamlit/streamlit/issues/3977)

Seems streamlit still has quite some gaps. The other notable one is that you cant sent css to an individual widget (which is kinda how react does it normally)... so you also cant just hack a native component to look like a toggle. 

So limited built in options and not a lot of ways to get around this.
